packages/backend/src/utils/Publisher.ts

```diff
@@ -22,17 +22,19 @@ export class Publisher<T> {
   constructor(
     private webview: Webview,
     private channel: Messages,
-    private getter: () => T,
+    private stateGetter: () => T,
   ) {}
 
-  notify(): void {
-    this.webview
-      .postMessage({
+  protected async notify(): Promise<void> {
+    try {
+      const data = this.stateGetter();
+      await this.webview.postMessage({
         id: this.channel,
-        body: this.getter(),
-      })
-      .catch((err: unknown) => {
-        console.error(`Something went wrong while emitting ${this.channel}: ${String(err)}`);
+        body: data,
       });
+    } catch (error) {
+      console.error(`Error publishing to ${this.channel}:`, error);
+      throw error; // Re-throw to allow error handling by caller
+    }
```

Comment:
At the time when we created the `Publisher` class, we made it so the notify was not a promise, as the caller often do not want to deal with a promise.

The idea is also to consider that the notify cannot really fails, so there is no need to catch the `error(s)`. The rare cases where it could fails would be:

- notify is called too early in the startup of podman-desktop and the webview is not yet defined
- notify is called during the shutdown of podman-desktop and the webview is being destroyed

IMHO those two edge case are not _worth_ the need effort to add a try catch every-time we want to call the `notify` function.